### ISSUE TITLE
Type returned by parseGitPatch is not ParsedPatchType

### ISSUE THREAD

**Issue by pbrisbin** [dherault/parse-git-patch#17]
I see `ParsedPatchType` defined, but the `parseGitPatch` function has no type signature, and the inferred type doesn't match:

```js
const parsed: ParsedPatchType = parseGitPatch("...")
```

```
Argument of type '{ files: ParsedPatchFileDataType[]; hash?: undefined; authorName?: undefined; authorEmail?: undefined; date?: undefined; message?: undefined; } | { files: ParsedPatchFileDataType[]; ... 4 more ...; message: string; }' is not assignable to parameter of type 'ParsedPatchType'.
      Type '{ files: ParsedPatchFileDataType[]; hash?: undefined; authorName?: undefined; authorEmail?: undefined; date?: undefined; message?: undefined; }' is not assignable to type 'ParsedPatchType'.
        Types of property 'hash' are incompatible.
          Type 'undefined' is not assignable to type 'string'.
```

It seems to disagree about `hash` in one of the paths through the function.

Currently working around it with `as`, but not sure if I'm opening myself up to issues later.

**Comment by dherault** [comment]
Thanks, I'll look it up

**Comment by dherault** [comment]
Fixed in v1.3.0
